pinot-common/src/main/java/org/apache/pinot/common/utils/ZkStarter.java

```diff
@@ -200,6 +202,17 @@ public void run() {
     }
   }
 
+  public static void closeAsync(ZkClient client) {
+    if (client != null) {
+      ZK_DISCONNECTOR.submit(() -> {
+        client.close();
```

Comment:
How long does this take if we run it in sync fashion?

Reply:
Please have a look at the attached flame graph.